brainscore_vision/nwb_integration/extract_nwb_data.py

```diff
@@ -0,0 +1,72 @@
+import json
+import h5py
+import logging
+import remfile
+
+from dandi.dandiapi import DandiAPIClient
+from pynwb import NWBHDF5IO
+from pynwb.file import NWBFile
+
+
+logger = logging.getLogger(__name__)
+
+### Load nwb file
+def load_nwb_file(nwb_file_path: str) -> NWBFile:
+    logger.info("Loading the NWB file ...")
+    io = NWBHDF5IO(nwb_file_path, "r") 
+    nwb_file = io.read()    
+    return nwb_file
+    
+def validate_nwb_file(nwb_file_path: str, dandiset_id: str = None) -> NWBFile:
+    '''
+    Check if the provided NWB file contains PSTH electrode data
+    '''
+    try:
+        nwb_file = load_nwb_file(nwb_file_path)
+    except:
+        try:
+            nwb_file = stream_from_dandi(dandiset_id, nwb_file_path)
+        except:
+            raise ValueError('Unable to load NWB file')
+
+    if nwb_file.electrodes and 'PSTHs_QualityApproved_ZScored_SessionMerged' in nwb_file.scratch.keys():
+        return nwb_file
+    else:
+        raise ValueError('Type of NWB file not accepted, needs to contain electrode data.')
+
+def generate_json_file(nwb_file: NWBFile, json_file_path: str) -> None:
+    '''
+    Extracts metadata from NWB file and writes to a JSON
+    '''
+    try:
+        scratch = nwb_file.scratch['PSTHs_QualityApproved_ZScored_SessionMerged'].description.split('[start_time_ms, stop_time_ms, tb_ms]: ')[-1]
+        array = scratch.strip('[]').split()
+    except:
+        raise ValueError('Unable to extract PSTHs_QualityApproved_ZScored_SessionMerged scratch data')
+    nwb_metadata = {'start_time_ms': array[0],
+                    'stop_time_ms': array[1],
+                    'tb_ms': array[2],
+                    'subject': nwb_file.subject.subject_id,
+                    'exp_name': nwb_file.session_id
```

Comment:
eventually we probably want to add `region` to this as well (coming from website), but just wanted to flag that here